File: 1725-number-of-rectangles-that-can-form-the-largest-square/1725-number-of-rectangles-that-can-form-the-largest-square.cpp
class Solution {
public:
    int countGoodRectangles(vector<vector<int>>& rectangles) {
        int mx = 0;
        for (auto&v : rectangles) {
            mx = max(mx, min(v[0], v[1]));
        }
        int count = 0;
        for (auto&v : rectangles) {
            if (min(v[0], v[1]) >= mx) count++;
        }
        return count;
    }
};

File: 1725-number-of-rectangles-that-can-form-the-largest-square/1725-number-of-rectangles-that-can-form-the-largest-square.py
class Solution:
    def countGoodRectangles(self, rectangles: List[List[int]]) -> int:
        mx = 0
        for [l, b] in rectangles:
            mx = max(mx, min(l, b))
        count = 0
        for [l, b] in rectangles:
            if min(l, b) >= mx:
                count += 1
        return count